official/transformer/transformer_main.py

```diff
@@ -245,145 +267,199 @@ def train_schedule(
     eval_results = estimator.evaluate(dataset.eval_input_fn)
-    print("Evaluation results (iter %d/%d):" % (i + 1, train_eval_iterations),
-          eval_results)
-
+    tf.logging.info("Evaluation results (iter %d/%d):" %
+                    (i + 1, train_eval_iterations))
+    tf.logging.info(eval_results)
+    benchmark_logger.log_evaluation_result(eval_results)
+
+    # The results from estimator.evaluate() are measured on an approximate
+    # translation, which utilize the target golden values provided. The actual
+    # bleu score must be computed using the estimator.predict() path, which
+    # outputs translations that are not based on golden values. The translations
+    # are compared to reference file to get the actual bleu score.
     if evaluate_bleu:
-      uncased_score, _ = evaluate_and_log_bleu(
-          estimator, bleu_writer, bleu_source, bleu_ref)
-      if bleu_threshold is not None and uncased_score > bleu_threshold:
+      uncased_score, cased_score = evaluate_and_log_bleu(
+          estimator, bleu_source, bleu_ref, vocab_file_path)
+
+      # Write actual bleu scores using summary writer and benchmark logger
+      global_step = get_global_step(estimator)
+      summary = tf.Summary(value=[
+          tf.Summary.Value(tag="bleu/uncased", simple_value=uncased_score),
+          tf.Summary.Value(tag="bleu/cased", simple_value=cased_score),
+      ])
+      bleu_writer.add_summary(summary, global_step)
+      bleu_writer.flush()
+      benchmark_logger.log_metric(
+          "bleu_uncased", uncased_score, global_step=global_step)
+      benchmark_logger.log_metric(
+          "bleu_cased", cased_score, global_step=global_step)
+
+      # Stop training if bleu stopping threshold is met.
+      if model_helpers.past_stop_threshold(bleu_threshold, uncased_score):
         bleu_writer.close()
         break
 
 
-def main(_):
-  # Set logging level to INFO to display training progress (logged by the
-  # estimator)
-  tf.logging.set_verbosity(tf.logging.INFO)
+def define_transformer_flags():
+  """Add flags and flag validators for running transformer_main."""
+  # Add common flags (data_dir, model_dir, train_epochs, etc.).
+  flags_core.define_base(multi_gpu=False, num_gpu=False, export_dir=False)
+  flags_core.define_performance(
+      num_parallel_calls=True,
+      inter_op=False,
+      intra_op=False,
+      synthetic_data=False,
+      max_train_steps=False,
+      dtype=False
+  )
+  flags_core.define_benchmark()
+
+  # Set flags from the flags_core module as "key flags" so they're listed when
+  # the '-h' flag is used. Without this line, the flags defined above are
+  # only shown in the full `--helpful` help text.
+  flags.adopt_module_key_flags(flags_core)
+
+  # Add transformer-specific flags
+  flags.DEFINE_enum(
+      name="params", short_name="mp", default="big",
+      enum_values=["base", "big"],
+      help=flags_core.help_wrap(
+          "Parameter set to use when creating and training the model."))
```

Comment:
I think its also worth mentioning all the individual param values under this umbrella as well.

Reply:
I left it out because the absl `-h` option shows the possible enum values. 

This is what the help text looks like:

      -mp,--params: <base|big>:
        Parameter set to use when creating and training the model.
        (default: 'big')

Reply:
What i am trying to say is param_set=base will populate other a, b and c param, which is not showing up the help txt.

Reply:
My bad, I completely misread that. Yes, I think it would be good to see the individual param values. Maybe not all of them, but at least the ones that change between the big and base parameter sets.

Reply:
Something like `setting params_set=big increases the default batch size, as well as the hidden_size, filter_size, and num_heads topology hyperparameters.\nSee transformer/model/model_params.py for details.`?

Reply:
Yes, like that. I've pushed the change that updates the help text